src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/OkHttpWebSocketWrapper.java

```diff
@@ -97,39 +102,61 @@ public void onMessage(WebSocket webSocket, ByteString bytes) {
 
         @Override
         public void onClosing(WebSocket webSocket, int code, String reason) {
-            onClose.invoke(code, reason);
+            boolean isOpen = false;
+            stateLock.lock();
+            try {
+                isOpen = startSubject.hasComplete();
+            } finally {
+                stateLock.unlock();
+            }
+            // Only call onClose if connection is open
+            if (isOpen) {
+                onClose.invoke(code, reason);
+            }
+
             try {
-                closeLock.lock();
+                stateLock.lock();
                 closeSubject.onComplete();
             }
             finally {
-                closeLock.unlock();
+                stateLock.unlock();
             }
-            checkStartFailure();
+            checkStartFailure(null);
         }
 
         @Override
         public void onFailure(WebSocket webSocket, Throwable t, Response response) {
             logger.error("WebSocket closed from an error: {}.", t.getMessage());
```

Comment:
And while I'm thinking about logging. I assume the following will produce more helpful output.

```suggestion
            logger.error("WebSocket closed from an error.", t);
```